### PR TITLE
Activity as Link from shared repository 

### PR DESCRIPTION
**Pull request by Maheshkale447**

### CODE REVIEW

**Review by JinendraGhodke — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNETUnitTest/SolutionTestsLib/RepositorySerializerTest.cs [hunk 1]
```diff
@@ -97,7 +97,7 @@ public void ConvertBFToString()
             // Please double verify if the increase in size make sense and is needed before changing this value of expected length            
             int lt = xml.Count(f => f == '<');
             int gt = xml.Count(f => f == '>');
-            Assert.IsTrue(xml.Length < 800, "Verify minimal xml is less than 800 bytes");   
+            Assert.IsTrue(xml.Length < 850, "Verify minimal xml is less than 850 bytes");
```
[the remark this answers is not in the document]
Comment (Maheshkale447) [reply Ginger/GingerCoreNETUnitTest/SolutionTestsLib/RepositorySerializerTest.cs:100]: Yes, type is added to XML, hence unit test case was failing, so had to change the length size

File: Ginger/GingerCoreCommon/Repository/RepositoryItemBase.cs [hunk 2]
```diff
@@ -45,6 +45,13 @@ public enum SerializationErrorType
         PropertyNotFound,
         SetValueException   // if type changed, and we can add more handling...
     }
+    public enum eType
```
[the remark this answers is not in the document]
Comment (Maheshkale447) [reply Ginger/GingerCoreCommon/Repository/RepositoryItemBase.cs:48]: done

File: Ginger/GingerCoreCommon/Repository/NewRepositorySerializer.cs [hunk 3]
```diff
@@ -377,12 +377,18 @@ private void xmlwriteObservableList(XmlTextWriter xml, string Name, IObservableL
             foreach (var v in list)
             {                
                 if (v is RepositoryItemBase)
-                {                    
+                {
+                    if (((RepositoryItemBase)v).IsLinkedItem)
+                    {
```
[the remark this answers is not in the document]
Comment (Maheshkale447) [reply Ginger/GingerCoreCommon/Repository/NewRepositorySerializer.cs:382]: Yes, so actions won't be added in XML

File: Ginger/GingerCoreCommon/WorkSpaceLib/GingerCoreCommonWorkSpace.cs [hunk 4]
```diff
@@ -46,5 +47,11 @@ public Solution Solution
             get;
             set;
         }
+
```
[the remark this answers is not in the document]
Comment (Maheshkale447) [reply Ginger/GingerCoreCommon/WorkSpaceLib/GingerCoreCommonWorkSpace.cs:50]: No, Solution repository object is not available on solution.cs,

File: Ginger/GingerCoreNET/WorkSpaceLib/WorkSpace.cs [hunk 5]
```diff
@@ -421,6 +425,8 @@ public bool OpenSolution(string solutionFolder, string encryptionKey = null)
 
                 Reporter.ToLog(eLogLevel.INFO, "Loading Solution- Creating Items Repository");
                 SolutionRepository = GingerSolutionRepository.CreateGingerSolutionRepository();
+                //TODO: to find better solution
+                GingerCoreCommonWorkSpace.Instance.SolutionRepository = this.SolutionRepository;
```
[the remark this answers is not in the document]
Comment (Maheshkale447) [reply Ginger/GingerCoreNET/WorkSpaceLib/WorkSpace.cs:429]: yes, added that line temporarily, removed now

**Pull request merged**